volatility3/cli/volshell/generic.py

```diff
@@ -344,67 +362,129 @@ def display_type(
                 volobject.vol.type_name, layer_name=self.current_layer, offset=offset
             )
 
-        if hasattr(volobject.vol, "size"):
-            print(f"{volobject.vol.type_name} ({volobject.vol.size} bytes)")
-        elif hasattr(volobject.vol, "data_format"):
-            data_format = volobject.vol.data_format
-            print(
-                "{} ({} bytes, {} endian, {})".format(
-                    volobject.vol.type_name,
-                    data_format.length,
-                    data_format.byteorder,
-                    "signed" if data_format.signed else "unsigned",
-                )
-            )
+        # add special case for pointer so that information about the struct the
+        # pointer is pointing to is shown rather than simply the fact this is a
+        # pointer object. The "dereference_count < 8" is to guard against loops
+        dereference_count = 0
+        while isinstance(volobject, objects.Pointer) and dereference_count < 8:
+            # before defreerencing the pointer, show it's information
+            print("   " * dereference_count, self._display_simple_type(volobject))
+
+            # check that we can follow the pointer before dereferencing and do not
+            # attempt to follow null pointers.
+            if volobject.is_readable() and volobject != 0:
+                # now deference the pointer and store this as the new volobject
+                volobject = volobject.dereference()
+                dereference_count = dereference_count + 1
+            else:
+                # if we aren't able to follow the pointers anymore then there will
+                # be no more information to display as we've already printed the
+                # details of this pointer
+                return
 
         if hasattr(volobject.vol, "members"):
```

Comment:
Great question - I'll dig into it.

Reply:
I've been exploring this issue and found that when dealing with actual objects or types with offsets, checking if it's an instance of `objects.AggregateType` works really well. 👍

However, I ran into difficulties when working with templates, such as when executing `dt('task_struct')` without any offsets. Both cases appear as ObjectTemplate types, making it challenging to distinguish between them with type checks:

```python
(layer_name) >>> type(self.context.symbol_space.get_type('symbol_table_name1!task_struct'))       
<class 'volatility3.framework.objects.templates.ObjectTemplate'>
(layer_name) >>> type(self.context.symbol_space.get_type('symbol_table_name1!int'))            
<class 'volatility3.framework.objects.templates.ObjectTemplate'>
```

I found `.vol.object_class` promising for getting the types:

```python

(layer_name) >>> self.context.symbol_space.get_type('symbol_table_name1!task_struct').vol.object_class
<class 'volatility3.framework.symbols.linux.extensions.task_struct'>
(layer_name) >>> self.context.symbol_space.get_type('symbol_table_name1!int').vol.object_class         
<class 'volatility3.framework.objects.Integer'>
```

However, `.vol.object_class` ends up as `abc.ABCMeta`, making `isinstance` checks infeasible as far as I understand. Which is kind of understandable since they're not actual instances.

```python
(layer_name) >>> type(self.context.symbol_space.get_type('symbol_table_name1!task_struct').vol.object_class) 
<class 'abc.ABCMeta'>
(layer_name) >>> type(self.context.symbol_space.get_type('symbol_table_name1!int').vol.object_class)          
<class 'abc.ABCMeta'>

```

I've looked for solutions and found some discussions like [this one on StackOverflow](https://stackoverflow.com/questions/69417027/how-to-typecheck-class-with-method-inserted-by-metaclass-in-python), but I'm still uncertain about the best approach.

It might be possible to adjust the templates to facilitate `isinstance` checks, but it's currently beyond my expertise. 😕 For now, I've resorted to using `hasattr` to differentiate between types, though it feels like a workaround:

```python

(layer_name) >>> hasattr(self.context.symbol_space.get_type('symbol_table_name1!task_struct'), "members")                  
True
(layer_name) >>> hasattr(self.context.symbol_space.get_type('symbol_table_name1!int'), "members")         
False

```

I'm open to suggestions or guidance on how to approach this more effectively. 🙃 Let me know if you have any good ideas!